flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerConfiguration.java

```diff
@@ -244,7 +244,12 @@ private static MemorySize getMaxTotalMem(
                                         ? MemorySize.MAX_VALUE
                                         : defaultWorkerResourceSpec
                                                 .getTotalMemSize()
-                                                .divide(defaultWorkerResourceSpec.getNumSlots())
-                                                .multiply(maxSlotNum));
+                                                // In theory, there is a possibility of long
```

Comment:
I think getMaxTotalCpu has similar issues. Such as 10 cores with 3 slots.

Reply:
`CPUResource` uses `BigDecimal` internally, while `MemorySize` is actually stored in `long`. IIUC, `getMaxTotalCpu` should be safe 🤔

Reply:
We set RoundingMode to Down in BigDecimal.devide(). So it will lose some precision.

Reply:
Fair enough. I have update this in the fix-up commit. PTAL again.